taxonomy/models.py

```diff
@@ -357,3 +357,88 @@ def __repr__(self):
             'unique_postings="{4!r} unique_companies={5!r}">'.format(
                 self.id, self.job, self.median_salary, self.median_posting_duration, self.unique_postings,
                 self.unique_companies)
+
+
+class Translation(TimeStampedModel):
+    """
+    Model to save translated descriptions.
+
+    .. no_pii:
+    """
+
+    source_model_name = models.CharField(
+        max_length=255,
+        help_text=_(
+            'The name of the model to which the source belongs e:g Course.'
+        )
+    )
+
+    source_model_field = models.CharField(
+        max_length=255,
+        help_text=_(
+            'The name of the source field to be translated e:g course description.'
+        )
+    )
+
+    source_record_identifier = models.CharField(
+        max_length=255,
+        help_text=_('The identifier of the source record e:g course key.')
+    )
+
+    source_text = models.TextField(
+        blank=True,
+        null=True,
+        help_text=_(
+            'The source text to be translated.'
+        )
+    )
+
+    source_language = models.CharField(
+        max_length=8,
+        help_text=_(
+            'The original language of the source text before translation e:g Spanish.'
+        )
+    )
+
+    translated_text = models.TextField(
+        blank=True,
+        null=True,
+        help_text=_(
+            'The translated source text.'
+        )
+    )
+
+    translated_text_language = models.CharField(
```

Comment:
I think we need to keep this `blank=True, null=True,` as well as we are applying these checks on `translated_text` field.

Reply:
Good catch, done.